src/java.base/share/classes/java/util/UUID.java

```diff
@@ -514,8 +514,6 @@ public int compareTo(UUID val) {
         // can simply be numerically compared as two numbers
         return (this.mostSigBits < val.mostSigBits ? -1 :
                 (this.mostSigBits > val.mostSigBits ? 1 :
-                 (this.leastSigBits < val.leastSigBits ? -1 :
-                  (this.leastSigBits > val.leastSigBits ? 1 :
-                   0))));
+                 Long.compare(this.leastSigBits, val.leastSigBits)));
```

Comment:
In this case Javadoc specifies only -1, 0 or 1 can be returned. `Long.compare` does not specify this but returns these values. Couldn't it cause any problems in the future if implementation of `Long.compare` is changed?

Does it make sense to use `Long.compare` for `mostSigBits` too?
```java
int mostSigBits = Long.compare(this.mostSigBits, val.mostSigBits);
return mostSigBits != 0 ? mostSigBits : Long.compare(this.leastSigBits, val.leastSigBits);
```

Reply:
This behavior is specified in `Comparator` itself, so is hardly to be ever changed, I think

Reply:
> This behavior is specified in `Comparator` itself, so is hardly to be ever changed, I think

Not quite. [`Comparator.compare` returns](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Comparator.html#compare(T,T)): “a negative integer, zero, or a positive integer as the first argument is less than, equal to, or greater than the second.” [`Comparable.compareTo`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Comparable.html#compareTo(T)) does the same. Yet Javadoc for `UUID` specifically requires `-1`, `0` or `1`, which is stricter. However, the implementation of `Long.compare` complies with this requirement even though it's not specified.

Probably, it makes sense to update the Javadoc for `UUID` and relax the requirement to “a negative integer, zero, or a positive integer.”

Reply:
Done

Reply:
I guess I didn't make myself clear. It wasn't a showstopper but it's still an issue.

This kind of change to Javadoc requires a CSR therefore it's better to separate it from this changeset which touches other files.

I see two options: 1) leave this unchanged and update implementation together with Javadoc; or 2) integrate the update to the implementation which you already have and update the Javadoc in a follow-up bug.

Reply:
The suggestion to compare `mostSigBits` using `Long.compare` if it looks reasonable still applies.

Reply:
Ok, let me revert JavaDoc and apply the suggestion